dali/pipeline/operators/reader/coco_reader_op_test.cc

```diff
@@ -0,0 +1,301 @@
+    240, 40, 45, 230, 61, 345, 115, 474, 183, 70, 107, 360, 73, 244, 99,
+    418, 110, 109, 172, 197, 210, 251, 50, 84, 147, 148, 127, 272, 53, 203,
+    154, 166, 49, 152, 110, 193, 160, 116, 139, 524, 82, 47, 152, 365, 3,
+    115, 119, 357, 212, 17, 30, 402, 37, 138, 23, 421, 121, 94, 197, 74, 64,
+    314, 125, 55, 59, 140, 76, 129, 173, 213, 132, 230, 106, 315, 219, 435,
+    193, 38, 56, 2, 89, 261, 25, 147, 228, 152, 126, 10, 195, 223, 357, 28,
+    163, 231, 305, 63, 416, 202, 76, 29, 150, 80, 129, 68, 510, 257, 10, 169,
+    258, 295, 33, 313, 131, 88, 142, 392, 205, 91, 75, 9, 132, 317, 120,
+    425, 2, 17, 200, 92, 162, 349, 128, 56, 5, 53, 107, 401, 154, 25, 103,
+    122, 75, 293, 129, 10, 262, 195, 47, 389, 131, 205, 163, 149, 183, 154,
+    144, 488, 139, 17, 137, 138, 82, 323, 57, 377, 33, 220, 216, 22, 268,
+    243, 128, 311, 39, 99, 215, 221, 61, 85, 214, 46, 93, 192, 187, 91, 184,
+    32, 12, 202, 203, 225, 171, 96, 207, 323, 64, 249, 106, 211, 254, 189,
+    223, 316, 89, 184, 130, 322, 115, 336, 74, 244, 22, 519, 70, 139, 189,
+    331, 84, 145, 181, 128, 189, 75, 152, 56, 408, 141, 26, 353, 217, 128, 69,
+    167, 238, 331, 233, 93, 166, 285, 23, 22, 55, 68, 140, 260, 44, 41, 231,
+    341, 180, 120, 91, 378, 208, 127, 112, 217, 24, 261, 160, 117, 94, 71, 86,
+    46, 171, 313, 115, 199, 18, 65, 34, 105, 145, 322, 113, 270, 5, 31, 135,
+    302, 337, 169, 76
+  };
+
+  std::vector<int> categories_ {
+    33, 34, 12, 20, 8, 34, 28, 49, 36, 70, 56, 23, 25, 24, 64, 1,
+    42, 44, 73, 72, 5, 39, 8, 10, 14, 75, 50, 22, 77, 71, 31, 63,
+    32, 70, 59, 27, 69, 74, 37, 14, 22, 45, 16, 60, 16, 78, 15, 30,
+    29, 58, 38, 25, 79, 28, 74, 47, 67, 28, 1, 27, 11, 25, 17, 39,
+    31, 16, 32, 75, 59, 72, 15, 58, 11, 18, 25, 72, 32, 44, 17, 45,
+    80, 77, 61, 68, 3, 20, 45, 70, 47, 2, 42, 73, 51, 64, 48, 19,
+    69, 70, 65, 29, 39, 68, 45, 15, 15, 48, 49, 44, 43, 12, 50, 15,
+    23, 54, 39, 62, 17, 6, 73, 58, 19, 41, 71, 61, 79, 32, 12, 27,
+    33, 1, 19, 76, 15, 18, 34, 17, 76, 39, 44, 51, 7, 44, 36, 10, 3,
+    32, 39, 46, 78, 38, 55, 68, 71, 29, 20, 80, 4, 66, 44, 1, 36, 54,
+    24, 23, 20, 51, 46, 11, 71, 46, 49, 16, 12, 24, 54, 33, 46, 16,
+    4, 64, 76, 2, 77, 17, 56, 5, 48, 1, 30, 19, 14, 5, 62, 31,
+  };
+};
+
+TEST_F(CocoReaderTest, MutuallyExclusiveOptions) {
+  Pipeline pipe(1, 1, 0);
+
+  pipe.AddOperator(
+    this->CocoReaderOpSpec()
+    .AddArg("shuffle_after_epoch", true)
+    .AddArg("stick_to_shard", true));
+
+  EXPECT_THROW(pipe.Build(this->Outputs()), std::runtime_error);
+}
+
+TEST_F(CocoReaderTest, SkipEmpty) {
+  Pipeline pipe(this->SmallCocoSize() - this->EmptyImages(), 1, 0);
+
+  pipe.AddOperator(
+    this->CocoReaderOpSpec().AddArg("skip_empty", true),
+    "coco_reader");
+
+  pipe.Build(this->Outputs());
+
+  ASSERT_EQ(
+    pipe.EpochSize()["coco_reader"],
+    this->SmallCocoSize() - this->EmptyImages());
+
+  DeviceWorkspace ws;
+  pipe.RunCPU();
+  pipe.RunGPU();
```

Comment:
Isn't `RunMixed()` required here? @klecki ?

Reply:
`RunMixed` is no more there